examples/histogram/src/main.cpp

```diff
@@ -68,25 +68,30 @@ void sparse_histogram(std::vector<uint64_t> &input) {
   std::sort(oneapi::dpl::execution::dpcpp_default,
             oneapi::dpl::begin(histogram_buf), oneapi::dpl::end(histogram_buf));
 
-  auto num_bins = std::transform_reduce(
-      oneapi::dpl::execution::dpcpp_default, oneapi::dpl::begin(histogram_buf),
-      oneapi::dpl::end(histogram_buf) - 1, oneapi::dpl::begin(histogram_buf) + 1, 1,
-      std::plus<int>(), std::not_equal_to<int>());
-
-  // Create new buffer to store the unique values and their count
-  sycl::buffer<uint64_t> histogram_values_buf{sycl::range<1>(num_bins)};
-  sycl::buffer<uint64_t> histogram_counts_buf{sycl::range<1>(num_bins)};
+  // Create new buffer to store the unique values and their count:
+  // used the size of input data instead of preevaluated count of baskets
+  // because reduce_by_segment required size of output data
+  // not less than size of input data.
+  sycl::buffer<uint64_t> histogram_values_buf{sycl::range<1>(N)};
+  sycl::buffer<uint64_t> histogram_counts_buf{sycl::range<1>(N)};
 
   sycl::buffer<uint64_t> _const_buf{sycl::range<1>(N)};
   std::fill(oneapi::dpl::execution::dpcpp_default,
             oneapi::dpl::begin(_const_buf), oneapi::dpl::end(_const_buf), 1);
 
+  auto histogram_values_buf_begin = oneapi::dpl::begin(histogram_values_buf);
+  auto histogram_counts_buf_begin = oneapi::dpl::begin(histogram_counts_buf);
+
   // Find the count of each value
-  oneapi::dpl::reduce_by_segment(
+  const auto result = oneapi::dpl::reduce_by_segment(
       oneapi::dpl::execution::dpcpp_default, oneapi::dpl::begin(histogram_buf),
       oneapi::dpl::end(histogram_buf), oneapi::dpl::begin(_const_buf),
-      oneapi::dpl::begin(histogram_values_buf),
-      oneapi::dpl::begin(histogram_counts_buf));
+      histogram_values_buf_begin,
+      histogram_counts_buf_begin);
+
+  assert(result.first - histogram_values_buf_begin == result.second - histogram_counts_buf_begin);
```

Comment:
I would move the line after `const auto num_bins =...` 

`assert(num_bins == result.second - histogram_counts_buf_begin);`